### PR TITLE
Event should avoid altering the input map.

### PR DESCRIPTION
**Pull request by piygoyal**
### Context

Event class currently just reuses the input map and alters it which is not ideal and can potentially fail if the input map is immutable.

### Checklist

- [] `./gradlew build` compiles code correctly
- [] Added new tests where applicable
- [] `./gradlew test` passes all tests
- [] Extended README or added javadocs where applicable
- [] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java [hunk 1]
```diff
@@ -38,11 +38,23 @@
...
-        this(new HashMap<>());
+        this(null, false);
     }
 
     public Event(Map<String, Object> attributes) {
-        this.attributes = attributes;
+        this(attributes, true);
+    }
+
+    public Event(Map<String, Object> attributes, boolean deepCopy) {
```
Comment (nickmahilani) [inline mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java]: We can avoid exposing the deepcopy in the Event constructor api as it is an internal detail that the user probably doesn't understand.

File: mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java [hunk 2]
```diff
@@ -38,11 +38,23 @@
     private final Map<String, Object> attributes;
 
     public Event() {
-        this(new HashMap<>());
+        this(null, false);
     }
 
     public Event(Map<String, Object> attributes) {
-        this.attributes = attributes;
+        this(attributes, true);
```
Comment (nickmahilani) [inline mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java]: The default here is hardcoded and this is the most commonly used constructor for enqueuing events to MRE

File: mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java (hunk 1 above)
Comment (piygoyal) [reply mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java]: yeah, we discussed this and it was addressed in a follow-up PR. https://github.com/Netflix/mantis-publish/pull/22

File: mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java (hunk 2 above)
Comment (piygoyal) [reply mantis-publish-core/src/main/java/io/mantisrx/publish/api/Event.java]: also addressed in the follow-up PR. https://github.com/Netflix/mantis-publish/pull/22

**Pull request merged**